### DISCUSSION TITLE
Correct config to try Llama 2 70b on this pc without crashing?

### DISCUSSION THREAD

**Discussion by ZeroCool22** [oobabooga/text-generation-webui#3292] [Q&A]
PC SPECS:

- GPU: 1080 TI. (11 vram)
- 32gb of RAM.
- CPU:  5900x.

Thx in advance.

**Comment by viperwasp** [comment]
You only have two possible options.

Run on GPTQ 4 Bit where you load as much as you can onto your 12GB and offset the rest to CPU. This will be extremely slow and I'm not sure your 11GB VRAM + 32GB of RAM is enough. I think your capped to 2 thread CPU performance. 

The better option if can manage it is to download the 70B model in GGML format. Maybe something like 4_K_M or 5_K_M. It loads into your regular RAM and offsets as much as you can manage onto your GPU. This can be slow however should be a lot faster. As it uses as many threads as you want for the CPU. 

Your limit might GGML 4_K_S actually for 70B models. Anymore may not load into your 43 total GB of ram. It may also be possible to do other tweaks etc. But this is the best I can recommend I'm not a pro user of any of this.

**Comment by ZeroCool22** [comment]
> You only have two possible options.
> 
> Run on GPTQ 4 Bit where you load as much as you can onto your 12GB and offset the rest to CPU. This will be extremely slow and I'm not sure your 11GB VRAM + 32GB of RAM is enough. I think your capped to 2 thread CPU performance.
> 
> The better option if can manage it is to download the 70B model in GGML format. Maybe something like 4_K_M or 5_K_M. It loads into your regular RAM and offsets as much as you can manage onto your GPU. This can be slow however should be a lot faster. As it uses as many threads as you want for the CPU.
> 
> Your limit might GGML 4_K_S actually for 70B models. Anymore may not load into your 43 total GB of ram. It may also be possible to do other tweaks etc. But this is the best I can recommend I'm not a pro user of any of this.

Yeah, you are right, maybe i will just pass and keep using GPT and Claude online, today early i tried to run the 70b (without knowing anything about the configs) and got a blue screen of death on Win. 10 about memory and restarted, **never ever i got a blue screen since i have this PC**, lol.

**Reply by viperwasp** [reply]
Yes 65-70B is hard to run. But you will be able to run 30-34B with your setup using GGML CPU only and even faster if you setup GGML with GPU acceleration. Even 32GB on it's own without your GPU is enough I would recommend GGML 5_K_M.

**Reply by viperwasp** [reply]
Oh and yes I know the 34B Llama2 is not out yet. It should eventually be released. I am hoping it does as it will become my primary model used at least models trained off of it will. I predict we will see some of the best models ever come from it.

**Reply by thesimpleone** [reply]
Try the 13B in the meantime.

**Comment by ZeroCool22** [comment]
> You only have two possible options.
> 
> Run on GPTQ 4 Bit where you load as much as you can onto your 12GB and offset the rest to CPU. This will be extremely slow and I'm not sure your 11GB VRAM + 32GB of RAM is enough. I think your capped to 2 thread CPU performance.
> 
> The better option if can manage it is to download the 70B model in GGML format. Maybe something like 4_K_M or 5_K_M. It loads into your regular RAM and offsets as much as you can manage onto your GPU. This can be slow however should be a lot faster. As it uses as many threads as you want for the CPU.
> 
> Your limit might GGML 4_K_S actually for 70B models. Anymore may not load into your 43 total GB of ram. It may also be possible to do other tweaks etc. But this is the best I can recommend I'm not a pro user of any of this.

![Screenshot_25](https://github.com/oobabooga/text-generation-webui/assets/13344308/0baab605-46c5-4946-848c-23bed09ef253)

Which one?

**Reply by viperwasp** [reply]
I am a little lost on this as well? All previous models did not have two modes. I usually use instruct mode. So I think the one without chat is best for instruct mode? Soon I will have to figure out the answer to this question. 70B GGML support was only just added today. So I am likely going to grab Freewilly Llama 2 70B GGML when it is quantized by "TheBloke" and other version of 70B Llama 2. Am I correct that the "chat" model is better for chat mode like role playing etc? But for instruct you should download the non-chat model? If anyone knows please let us know.

**Reply by missionfloyd** [reply]
As I understand it, the chat version is for chat/instruct/whatever and is heavily censored, and the other is meant to be a base for finetuning and is uncensored.

**Reply by thesimpleone** [reply]
The base model is foundation model and isn't fine-tuned for specific tasks.  It would be used if you had a project that you want to further fine tune to your own specifications.    The chat model is optimized for dialogue. 

If you read the papers they help to better solidify their purpose.    Ideally start out with the chat model unless you plan on your own fine tuning.  

https://ai.meta.com/research/publications/llama-2-open-foundation-and-fine-tuned-chat-models/
